packages/webcomponents/src/components/payments-list/payments-list-core.tsx

```diff
@@ -0,0 +1,122 @@
+import { Component, Host, h, Prop, State, Watch, Event, EventEmitter } from '@stencil/core';
+import { PagingInfo, Payment, pagingDefaults } from '../../api';
+import { MapPaymentStatusToBadge, formatCurrency, formatDate, formatTime } from '../../utils/utils';
+
+@Component({
+  tag: 'payments-list-core',
+  styleUrl: 'payments-list.scss',
+})
+
+export class PaymentsListCore {
+  @Prop() getPayments: Function;
+  @State() payments: Payment[] = [];
+  @State() loading: boolean = true;
+  @State() errorMessage: string;
+  @State() paging: PagingInfo = pagingDefaults;
+  @State() params: any;
+  @Event({
+    eventName: 'payment-row-clicked',
+
+    bubbles: true,
+  }) rowClicked: EventEmitter<Payment>;
+
+  componentWillLoad() {
+    if (typeof this.getPayments === 'function') {
+      this.fetchData();
+    }
+  }
+
+  @Watch('params')
+  @Watch('getPayments')
+  updateOnPropChange() {
+    this.fetchData();
+  }
+
+  handleClickPrevious = (beforeCursor: string) => {
+    const newParams: any = { ...this.params };
+    delete newParams.after_cursor;
+    this.params = ({ ...newParams, before_cursor: beforeCursor });
+  };
+
+  handleClickNext = (afterCursor: string) => {
+    const newParams: any = { ...this.params };
+    delete newParams.before_cursor;
+    this.params = ({ ...newParams, after_cursor: afterCursor });
+  };
+
+  fetchData(): void {
+    this.loading = true;
+
+    if (typeof this.getPayments === 'function') {
```

Comment:
Similar to my other comment in this file - I think this logic is repeated here when perhaps it doesn't need to be?